src/test/java/com/google/cloud/spanner/jdbc/it/ITJdbcPgNumericTest.java

```diff
@@ -0,0 +1,271 @@
+import com.google.cloud.spanner.IntegrationTestEnv;
+import com.google.cloud.spanner.Value;
+import com.google.cloud.spanner.connection.ConnectionOptions;
+import com.google.cloud.spanner.jdbc.JdbcSqlExceptionFactory.JdbcSqlExceptionImpl;
+import com.google.cloud.spanner.testing.EmulatorSpannerHelper;
+import com.google.cloud.spanner.testing.RemoteSpannerHelper;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@Category(IntegrationTest.class)
+@RunWith(JUnit4.class)
+public class ITJdbcPgNumericTest {
+
+  @ClassRule public static IntegrationTestEnv env = new IntegrationTestEnv();
+
+  private static final Duration OPERATION_TIMEOUT = Duration.ofMinutes(10);
+  private static RemoteSpannerHelper testHelper;
+  private static Database database;
+  private static String url;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    assumeFalse(
+        "PgNumeric is not supported in the emulator", EmulatorSpannerHelper.isUsingEmulator());
+    testHelper = env.getTestHelper();
+    final String projectId = testHelper.getInstanceId().getProject();
+    final String instanceId = testHelper.getInstanceId().getInstance();
+    final String databaseId = testHelper.getUniqueDatabaseId();
+    final DatabaseAdminClient databaseAdminClient = testHelper.getClient().getDatabaseAdminClient();
+    final Database databaseToCreate =
+        databaseAdminClient
+            .newDatabaseBuilder(DatabaseId.of(projectId, instanceId, databaseId))
+            .setDialect(Dialect.POSTGRESQL)
+            .build();
+    final String host = SpannerTestHost.getHost();
+
+    database =
+        databaseAdminClient
+            .createDatabase(databaseToCreate, Collections.emptyList())
+            .get(OPERATION_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+    url = "jdbc:cloudspanner://" + host + "/" + database.getId() + "?dialect=postgresql";
+  }
+
+  @AfterClass
+  public static void teardown() {
+    if (database != null) {
+      database.drop();
+    }
+    ConnectionOptions.closeSpanner();
+  }
+
+  @Test
+  public void testResultSet() throws SQLException {
+    assumeFalse(
+        "PgNumeric is not supported in the emulator", EmulatorSpannerHelper.isUsingEmulator());
+    final String table = testHelper.getUniqueDatabaseId();
+    final String positiveBigNumeric =
+        String.join("", Collections.nCopies(131072, "1"))
+            + "."
+            + String.join("", Collections.nCopies(16383, "2"));
```

Comment:
Just for understanding: Do the numbers  `131072` and `16383` hold any special significance ?
`String.join("",` Collections.nCopies(131072, "1"))
            + "."
            + String.join("", Collections.nCopies(16383, "2"));`

Reply:
Yes, they represent the maximum number of digits before/after the decimal point in a PostgreSQL numeric: https://www.postgresql.org/docs/current/datatype-numeric.html